controllers/component_build_controller_common.go

```diff
@@ -0,0 +1,263 @@
+		{Namespace: buildServiceNamespaceName, Name: buildPipelineSelectorResourceName},
+	}
+
+	for _, pipelineSelectorKey := range pipelineSelectorKeys {
+		if err := r.Client.Get(ctx, pipelineSelectorKey, pipelineSelector); err != nil {
+			if !errors.IsNotFound(err) {
+				return nil, nil, err
+			}
+			// The config is not found, try the next one in the hierarchy
+		} else {
+			pipelineSelectors = append(pipelineSelectors, *pipelineSelector)
+		}
+	}
+
+	if len(pipelineSelectors) > 0 {
+		pipelineRef, pipelineParams, err := pipelineselector.SelectPipelineForComponent(component, pipelineSelectors)
+		if err != nil {
+			return nil, nil, err
+		}
+		if pipelineRef != nil {
+			return pipelineRef, pipelineParams, nil
+		}
+	}
+
+	// Fallback to the default pipeline
+	return &tektonapi.PipelineRef{
+		Name:   defaultPipelineName,
+		Bundle: defaultPipelineBundle,
+	}, nil, nil
+}
+
+func (r *ComponentBuildReconciler) ensurePipelineServiceAccount(ctx context.Context, namespace string) (*corev1.ServiceAccount, error) {
+	pipelinesServiceAccount := &corev1.ServiceAccount{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: "pipeline", Namespace: namespace}, pipelinesServiceAccount)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			r.Log.Error(err, fmt.Sprintf("Failed to read service account %s in namespace %s", buildPipelineServiceAccountName, namespace))
+			return nil, err
+		}
+		// Create service account for the build pipeline
+		buildPipelineSA := corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      buildPipelineServiceAccountName,
+				Namespace: namespace,
+			},
+		}
+		if err := r.Client.Create(ctx, &buildPipelineSA); err != nil {
+			r.Log.Error(err, fmt.Sprintf("Failed to create service account %s in namespace %s", buildPipelineServiceAccountName, namespace))
+			return nil, err
+		}
+		return r.ensurePipelineServiceAccount(ctx, namespace)
+	}
+	return pipelinesServiceAccount, nil
+}
+
+func (r *ComponentBuildReconciler) linkImageRegistrySecretToServiceAccount(ctx context.Context, imageRepo, imageRepoSecretName string, serviceAccount *corev1.ServiceAccount) (bool, error) {
+	log := r.Log
+
+	// Ensure that image repo secret is annotated to make it respected by Tekton
+
+	// quay.io/userlogin/image:tag => https://quay.io
+	expectedTektonAnnotationValue := "https://" + strings.Split(imageRepo, "/")[0]
+
+	dockerSecret := corev1.Secret{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: imageRepoSecretName, Namespace: serviceAccount.Namespace}, &dockerSecret)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Secret %s is missing", imageRepoSecretName))
+		return false, boerrors.NewBuildOpError(boerrors.EComponentImageRegistrySecretMissing, err)
+	}
+	if dockerSecret.Annotations == nil {
+		dockerSecret.Annotations = map[string]string{}
+	}
+	// We can always use 0 index because the component uses only one image repository
+	if dockerSecret.Annotations["tekton.dev/docker-0"] != expectedTektonAnnotationValue {
```

Comment:
this annotation should not be used -> it's only for [basic-auth](https://tekton.dev/docs/pipelines/auth/#configuring-basic-auth-authentication-for-docker) -> expects username and  password, image controller provides dockerauth file directly.

Reply:
ok, that's simplifies stuff.